addons/website/models/ir_http.py

```diff
@@ -145,12 +145,12 @@ def _register_website_track(cls, response):
             return False
 
         template = False
-        if hasattr(response, 'qcontext'):  # classic response
+        if hasattr(response, '_cached_page'):
+            website_page, template = response._cached_page, response._cached_template
+        elif hasattr(response, 'qcontext'):  # classic response
             main_object = response.qcontext.get('main_object')
             website_page = getattr(main_object, '_name', False) == 'website.page' and main_object
             template = response.qcontext.get('response_template')
```

Comment:
the inversion of priorities could perhaps use some explanation in the commit? was it a bug before? (didn't check thoroughly)

Reply:
No bug before...
but since we always use an Odoo Response now. When you receive a cached version, it will have a _qcontext attr too.
https://github.com/odoo/odoo/pull/72599/files#diff-f444d9b99919dbdbf8a57e1492478cb518cd2f78ec07dfedebff81140210671bR300

So it allow to check if cached version before to fallback on default usecase